src/classes/PassageError/PassageError.ts

```diff
@@ -1,28 +1,32 @@
-import { ResponseError } from '../generated';
+import { ResponseError } from '../../generated';
 
 /**
  * Passage Class
  */
 export class PassageError extends Error {
     override name: 'PassageError' = 'PassageError';
     readonly statusCode: number | undefined;
-    readonly error: string | undefined;
+    readonly errorCode: string | undefined;
     readonly message: string;
 
+    /** @deprecated use errorCode instead */
+    readonly error: string | undefined;
+
     /**
      * Initialize a new PassageError instance.
-     * @param {string} message friendly message
+     * @param {string} message friendly error message
+     * @param {string} errorCode error code from Passage Backend
      * @param {ResponseError} err error from http request
      */
-    constructor(message: string, err?: ResponseError) {
+    constructor(message: string, errorCode?: string, err?: ResponseError) {
```

Comment:
nitpick: should this have a deprecation notice as well if we intend on only using the factory method to initialize `PassageError`? it could also help prevent incorrect usage of this class

i imagine this would become private and current invokers of this would switch to a standard `Error`

Reply:
We'll always want to expose the `PassageError` class for type-safe error handing even if we never construct it directly outside a factory method.

Reply:
sorry i may have miscommunicated; could the `PassageError` class still be exposed while its constructor is private?

Reply:
Private constructors are not a feature in javascript

Reply:
TS allows for a `private constructor(...)`, which the TS language server will flag as an error if called `new Class(...)`

![image](https://github.com/user-attachments/assets/6bd77769-4cb7-477a-bfc8-6f0b56101486)

are you referring to something other than this interaction?